AppBuilder/platform/ABDataCollection.js

```diff
@@ -531,101 +459,40 @@ module.exports = class ABDataCollection extends ABDataCollectionCore {
-                  if (this.__filterDatasource.isValid(entry)) {
-                     validEntries.push(entry);
-                  }
-               });
-
-               // load our valid entries:
-               // ! This call prevents this from returning found data @achoobert
-               // ? Why would this ever be nessicary?
-               this.processIncomingData(validEntries);
-
-               // we can start working on this data now
-               // NOTE: resolve() should be done in .processIncomingData() now
-               // resolve(validEntries);
-            })
-            .then((validEntries) => {
-               // ? is it really nessicary to call the remote here? @achoobert
-               // However, this local data might be out of date
-               // with the server.  So let's spawn a remote
-               // lookup in the background:
-
-               var modelRemote = model.remote();
-
-               // reset the context on the Model so any data updates get sent to this
-               // DataCollection
-               // NOTE: we only do this on loadData(), other operations should be
-               // received by the related Objects.
-               modelRemote.contextKey(ABDataCollectionCore.contextKey());
-               modelRemote.contextValues({
-                  id: this.id,
-                  verb: "refresh",
-               });
-
-               // id: the datacollection.id
-               // verb: tells our ABRelay.listener why this remote lookup was called.
-
-               // initiate the request:
-               modelRemote.findAll(cond);
-
-               // return valid entries:
-               return validEntries;
-            });
-      } else {
-         //  We have not been initialized yet, so we need to
-         //  request our data from the remote source()
-         var modelRemote = model.remote();
-
-         // reset the context on the Model so any data updates get sent to this
-         // DataCollection
-         // NOTE: we only do this on loadData(), other operations should be
-         // received by the related Objects.
-         modelRemote.contextKey(ABDataCollectionCore.contextKey());
-         modelRemote.contextValues({
-            id: this.id,
-            verb: "uninitialized",
-         });
-         // id: the datacollection.id
-         // verb: tells our ABRelay.listener why this remote lookup was called.
-
-         // initiate the request:
-         return modelRemote.findAll(cond);
-         // note:  our ABRelay.listener will take incoming data and call:
-         // this.processIncomingData()
+   async platformFind(model, cond) {
+      const myModel = model || this.datasource.model();
+      const modelRemote = myModel.remote();
+      const contextValues = {
+         id: this.id,
+         verb: "refresh",
+      };
+      if (this.bootState !== "initialized") {
+         contextValues.verb = "uninitialized";
+         await this.processIncomingData(
```

Comment:
For some reason this logic has always been difficult for me to grok; is this correct?
You are double awaiting: `findAll` is: "get the fully populated relation data from server"

Should we have some sort of timeout anywhere in this chain? 
I can't actually find who/what would ever call this function?